scripts/check-changelog.py

```diff
@@ -61,28 +72,29 @@
               % ' or '.join(repr(line) for line in acceptable_lines))
         found_problem = True
 
-    if all_versions != sorted(all_versions, reverse=True):
-        print('You edited old release notes and changed the ordering!')
-        found_problem = True
+    for actual, expected in zip(v_nums, sorted(v_nums, reverse=True)):
+        if actual != expected:
+            print('Version order: expected %s, got %s' % (expected, actual))
+            found_problem = True
 
-    v_nums = [v.split(' - ')[0] for v in all_versions]
     duplicates = sorted(set(v for v in v_nums if v_nums.count(v) > 1))
     if duplicates:
         plural = 's have' if len(duplicates) > 1 else ' has'
+        entries = ', '.join(repr(version_string(v)) for v in duplicates)
```

Comment:
Minor: in CPython, it’s marginally more efficient to call `.join()` on a list than a generator, because CPython immediately casts the argument to `join()` to a list – we cut out the intermediate generator comprehension.

As a result, I always reach for a list when calling `join()` unless I have a good reason not to – although in this case, I’ll admit performance isn’t an issue. 😉